localstack/services/generic_proxy.py

```diff
@@ -147,9 +147,10 @@ def forward(self, method):
             # copy headers and return response
             self.send_response(response.status_code)
             for header_key, header_value in iteritems(response.headers):
-                if header_key.lower() != 'Content-Length'.lower():
+                if header_key.lower() != 'Content-Length'.lower() or method == 'HEAD':
                     self.send_header(header_key, header_value)
-            self.send_header('Content-Length', '%s' % len(response.content))
+            if method != 'HEAD':
```

Comment:
how about this? This generalizes the cases. Any content-length header is passed through and if none is found len(response.content) is added.

```
content_length_sent = False
for header_key, header_value in iteritems(response.headers):
    self.send_header(header_key, header_value)
    content_length_sent = content_length_sent or header_key.lower() == 'content-length'
if not content_length_sent:
    self.send_header('Content-Length', '%s' % len(response.content))
```